contributors/design-proposals/storage/attacher-detacher-refactor-for-local-storage.md

```diff
@@ -0,0 +1,265 @@
+---
+
+title: Attacher/Detacher refactor for local storage
+
+authors:
+- "@NickrenREN"
+
+owning-sig: sig-storage
+
+participating-sigs:
+  - nil
+
+reviewers:
+  - "@msau42"
+  - "@jsafrane"
+
+approvers:
+  - "@jsafrane"
+  - "@msau42"
+  - "@saad-ali"
+
+editor: TBD
+
+creation-date: 2018-07-30
+
+last-updated: 2018-07-30
+
+status: provisional
+
+---
+
+## Table of Contents
+ * [Table of Contents](#table-of-contents)
+* [Summary](#summary)
+* [Motivation](#motivation)
+    * [Goals](#goals)
+    * [Non-Goals](#non-goals)
+* [Proposal](#proposal)
+* [Implementation](#implementation)
+    * [Volume plugin interface change](#volume-plugin-interface-change)
+    * [MountVolume/UnmountDevice generation function change](#MountVolume/UnmountDevice-generation-function-change)
+    * [Volume plugin change](#volume-plugin-change)
+* [Future](#future)
+
+## Summary
+
+Today, the workflow for a volume to be used by pod is:
+- attach a remote volume to the node instance (if it is attachable)
+- wait for the volume to be attached (if it is attachable)
+- mount the device to a global path (if it is attachable)
+- mount the global path to a pod directory
+
+It is ok for remote block storage plugins which have a remote attach api,such as `GCE PD`, `AWS EBS`
+and remote fs storage plugins such as `NFS`, and `Cephfs`.
+But it is not so good for plugins which need local attach such as `fc`, `iscsi` and `RBD`.
+It is not so good for local storage neither which is not attachable but needs `MountDevice`
```

Comment:
One more use case is for CSI, which separates MountDevice from Attach/Detach

Reply:
added, thanks

Reply:
talked with @msau42  offline.
Regarding CSI, Jan will find a way to avoid attaching operation for plugins which just need to mount device , but do not support attach,  such as local storage.
But this refactor will not benefit this.
/cc @jsafrane